src/tests/ftest/control/dmg_pool_query_ranks.py

```diff
@@ -0,0 +1,161 @@
+        finally:
+            self.dmg.exit_status_exception = True
+
+    def test_pool_query_ranks_mgmt(self):
+        """Test the state of ranks after excluding and reintegrate them
+
+        Test Description:
+            Create a pool with some engines exclude them one by one and check the consistency of the
+            list of enabled and disabled ranks.  Then, reintegrate them and check the consistency of
+            the list of enabled and disabled ranks.
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=vm
+        :avocado: tags=dmg,pool_query,pool_query_ranks
+        :avocado: tags=pool_query_ranks_mgmt
+        """
+        self.log.info("Tests of pool query with ranks state when playing with ranks")
+
+        enabled_ranks = [0, 1, 2]
+        disabled_ranks = []
+        for rank in [1, 0, 2]:
+            self.log.debug("Excluding rank %d", rank)
+            self.dmg.pool_exclude(self.pool.uuid, rank)
+            enabled_ranks.remove(rank)
+            disabled_ranks = sorted(disabled_ranks + [rank])
+
+            self.log.debug("Checking enabled ranks state information")
+            data = self.dmg.pool_query(self.pool.identifier, show_enabled=True)
+            self.assertListEqual(data['response']['enabled_ranks'], enabled_ranks,
+                    "Invalid enabled_ranks field: want={}, "
+                    "got={}".format(enabled_ranks, data['response']['enabled_ranks']))
+
+            self.log.debug("Checking disabled ranks state information")
+            data = self.dmg.pool_query(self.pool.identifier, show_disabled=True)
+            self.assertListEqual(data['response']['disabled_ranks'], disabled_ranks,
+                    "Invalid disabled_ranks field: want={}, "
+                    "got={}".format(disabled_ranks, data['response']['disabled_ranks']))
+
+        for rank in [2, 0, 1]:
+            self.log.debug("Reintegrating rank %d", rank)
+            self.dmg.exit_status_exception = False
+            # NOTE Reintegrating ranks could not be done immediately => looping with timeout until
+            # the eviction is done.
+            timeout = 60
+            try:
+                while timeout > 0:
+                    rc = self.dmg.pool_reintegrate(self.pool.uuid, rank)
+                    if rc.exit_status == 0:
+                        break
+                    self.assertIn('DER_BUSY', str(rc.stdout),
+                            "Pool reintegration failed: {}".format(str(rc.stdout)))
+                    self.log.debug('Pool reintegration of rank=%d failed: msg="Resource busy", '
+                            'timeout=%d', rank, timeout)
+                    time.sleep(1.)
+                    timeout -= 1
+            finally:
+                self.dmg.exit_status_exception = True
+            self.assertNotEqual(timeout, 0, "Rank {} could not be reintegrated".format(rank))
```

Comment:
Do not understand your remarks. The timeout is not for the command itself, but for the number of attempts to run it sucessfully. I am going to try to remove this mechanism with using `TestPool.wait_for_rebuild()` function (thanks to the remark of @shimizukko)

Reply:
Ah, I misread `timeout -= 1` as `timeout = -1`. If this is the usage, I would suggest naming it something like `retries` or `attempts`, since `timeout` has other connotations.